base_classes/NXobject.nxdl.xml

```diff
@@ -27,8 +27,43 @@
 	name="NXobject" 
 	type="group">
 	<doc>
-		This is the base object of NeXus
+        This is the base object of NeXus. The groups and fields contained 
+        within this file are allowed to be present in any derived base class.
+
+        If nameType="partial", the placeholders (e.g., FIELDNAME or GROUPNAME)
+        can be replaced by the name of any object (field or group, 
+        respectively) that exists within the same group.
 	</doc>
-        <!--attribute name="name"><doc>name of instance</doc></attribute-->
+    <group type="NXcollection" minOccurs="0" />
+    <group type="NXdata" minOccurs="0" />
+    <group type="NXgeometry" minOccurs="0" />
+    <group type="NXlog" minOccurs="0" />
+    <group type="NXnote" minOccurs="0" />
+    <group type="NXparameters" minOccurs="0" />
+    <group type="NXtransformations" minOccurs="0" />
+    <group name="GROUPNAME_log" type="NXlog" nameType="partial">
```

Comment:
Should these be partial nameTypes optional too as their presence is not required if the instances are present.

Reply:
I think all partial names are automatically optional. The whole point is that they are an optional supplement any other field in the same group. If an application definition wants to require one, it needs to specify the name exactly.

Reply:
Being in a base class actually also means that it is optional (by default). As Ray said, if needed, it can be made required in an AppDef. Note that optionality is not connected to nameType that only specifies the naming convention rule for inheritance. As an AppDef can require INSTRUMENT with nameType=any, it can also require GROUPNAME_log with nameType=partial. That would only mean that NXlog instances of xxx_log, yyy_log, etc. all belong to this concept.